lib/tasks/deploy.rake

```diff
@@ -0,0 +1,84 @@
+require "English"
+desc "Push the commit at the tip of the current branch to the dev environment"
+task push_to_dev: :environment do
+  puts "Getting Git commit..."
+  git_hash = `git rev-parse HEAD`.chomp
+  $CHILD_STATUS.success? or raise
+
+  puts "Will use Git commit: #{git_hash}"
+  puts "Checking commit has been pushed..."
+
+  resp = Net::HTTP.get_response URI("https://api.github.com/repos/alphagov/forms-product-page/commits/#{git_hash}")
+  if resp.code != "200"
+    puts "Commit has not been pushed. The pipelines will not be able to find it."
+    puts "Push the commit and run the command again."
+    raise
+  end
+  puts "The commit is present in the remote"
+
+  push_commit_to_dev(git_hash)
+end
+
+desc "Push the commit at the tip of origin/main to the dev environment"
+task reset_dev_to_main: :environment do
+  puts "Updating Git remotes"
+  sh "git", "fetch", verbose: false
+
+  git_hash = `git rev-parse origin/main`.chomp
+  $CHILD_STATUS.success? or raise
+
+  puts "Will use Git commit from tip of origin/main: #{git_hash}"
+
+  push_commit_to_dev(git_hash)
+end
+
+def push_commit_to_dev(git_hash)
+  json_input = {
+    name: "forms-product-page-image-builder",
+    sourceRevisions: [
+      {
+        actionName: "get-forms-product-page",
+        revisionType: "COMMIT_ID",
+        revisionValue: git_hash,
+      },
+    ],
+    variables: [
+      {
+        name: "tag_prefix",
+        value: "dev-",
+      },
+    ],
+  }
+
+  puts ""
+  puts "Starting pipeline 'forms-product-page-image-builder' in the 'deploy' account"
+  pp json_input
+
+  cli_input = JSON.dump(json_input)
+  command_args = [
+    "aws",
+    "codepipeline",
+    "start-pipeline-execution",
+    "--name",
+    "forms-product-page-image-builder",
+    "--cli-input-json",
+    cli_input,
+  ]
+  sh_aws(*command_args)
+end
+
+def sh_aws(*cmd)
+  if ENV.key?("AWS_VAULT") && (ENV["AWS_VAULT"] != "forms-deploy-support")
+    ENV.delete("AWS_VAULT")
+  end
+
+  all_args = ["gds", "aws", "forms-deploy-support", "--", *cmd]
+  system(*all_args)
+end
+
+def run(*cmd, &block)
+  escaped_cmd = cmd.map { |c| Shellwords.shellescape(c) }.join(" ")
+  stdout, stderr, status = Open3.capture3(escaped_cmd)
+  ok = status.zero?
+  block.call(ok, stdout, stderr)
+end
```

Comment:
This method doesn't seem to be used anywhere? 🤔

Reply:
Correct, this is the result of me doing haphazard changes in response to PR feedback and not paying enough attention. I gotta get better at writing and reviewing actual code I suppose :joy: